src/annoylib.h

```diff
@@ -296,81 +252,15 @@ struct Angular {
     // want to calculate (a/|a| - b/|b|)^2
     // = a^2 / a^2 + b^2 / b^2 - 2ab/|a||b|
     // = 2 - 2cos
-    T pp = 0, qq = 0, pq = 0;
-    for (int z = 0; z < f; z++, x++, y++) {
-      pp += (*x) * (*x);
-      qq += (*y) * (*y);
-      pq += (*x) * (*y);
-    }
+    T pp = dot(x, x, f), qq = dot(y, y, f), pq = dot(x, y, f);
```

Comment:
I wonder if rewriting it this way will be slower because of cache locality?

Reply:
hm yeah actually now that I'm looking at the AVX code, it definitely benefits from doing all of those at the same time (can load into a vector only twice, not six times)

let me add another specialized function that computes the dot products u*u, v*v and u*v at the same time – would be faster